llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -684,9 +686,56 @@ PrintIRInstrumentation::~PrintIRInstrumentation() {
   assert(ModuleDescStack.empty() && "ModuleDescStack is not empty at exit");
 }
 
-void PrintIRInstrumentation::pushModuleDesc(StringRef PassID, Any IR) {
+static SmallString<32> getIRFileDisplayName(Any IR) {
+  SmallString<32> Result;
+  raw_svector_ostream ResultStream(Result);
   const Module *M = unwrapModule(IR);
-  ModuleDescStack.emplace_back(M, getIRName(IR), PassID);
+  ResultStream << stable_hash_combine_string(M->getName());
```

Comment:
I suppose it makes sense to format the hashes as hex, so every filename has the same length